avs-aggregator/aggregator.go

```diff
@@ -166,10 +168,29 @@ func (agg *Aggregator) Start(ctx context.Context) error {
 	agg.logger.Infof("Starting aggregator rpc server.")
 	go agg.startServer(ctx)
 
-	sub, err := agg.substrateClient.RPC.Chain.SubscribeNewHeads()
+	var sub *gsrpcrpcchain.NewHeadsSubscription
+	var err error
+	const maxRetries = 5
+	const retryDelay = time.Minute
+
+	// Loop to retry subscription on error
+	for attempt := 0; attempt < maxRetries; attempt++ {
+		sub, err = agg.substrateClient.RPC.Chain.SubscribeNewHeads()
+		if err == nil {
+			break // Successfully subscribed, exit loop
+		}
+
+		agg.logger.Error("Failed to get new head from substrate, retrying...", "err", err, "attempt", attempt+1)
+		select {
+		case <-ctx.Done():
+			return nil
+		case <-time.After(retryDelay): // Wait before retrying
```

Comment:
Better for this would to use 

time.NewTicker(retryDelay) functionality native to Go but this works ok, up to you to change

Reply:
I don't need a ticker... I don't need it to ping periodically... I need it just once per attempt and I need the period to start after the attempt has completed unsuccessfully...